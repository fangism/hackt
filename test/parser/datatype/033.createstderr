
Persistent Object Manager text dump: 
	i	addr		type		arg	head	tail	size
	0	##ADDR##	________	0	#HEAD#	#TAIL#	0
	1	##ADDR##	module__	0	#HEAD#	#TAIL#	##SIZE##
	2	##ADDR##	namespc_	0	#HEAD#	#TAIL#	24
	3	##ADDR##	datadefn	0	#HEAD#	#TAIL#	120
	4	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	5	##ADDR##	dintplch	0	#HEAD#	#TAIL#	25
	6	##ADDR##	datatprf	0	#HEAD#	#TAIL#	20
	7	##ADDR##	datadefn	0	#HEAD#	#TAIL#	60
	8	##ADDR##	dprmlst_	0	#HEAD#	#TAIL#	16
	9	##ADDR##	cnstpi__	0	#HEAD#	#TAIL#	12
	10	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	11	##ADDR##	dboolplh	0	#HEAD#	#TAIL#	25
	12	##ADDR##	datatprf	0	#HEAD#	#TAIL#	20
	13	##ADDR##	datadefn	0	#HEAD#	#TAIL#	61
	14	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	15	##ADDR##	dboolplh	0	#HEAD#	#TAIL#	27
	16	##ADDR##	datatprf	0	#HEAD#	#TAIL#	20
	17	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	18	##ADDR##	dboolplh	0	#HEAD#	#TAIL#	26
	19	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	20	##ADDR##	dboolplh	0	#HEAD#	#TAIL#	26
	21	##ADDR##	datainst	0	#HEAD#	#TAIL#	24
	22	##ADDR##	dboolplh	0	#HEAD#	#TAIL#	27
	23	##ADDR##	crnglst_	0	#HEAD#	#TAIL#	20
	24	##ADDR##	datatprf	0	#HEAD#	#TAIL#	20
	25	##ADDR##	dprmlst_	0	#HEAD#	#TAIL#	16
	26	##ADDR##	cnstpi__	0	#HEAD#	#TAIL#	12
	27	##ADDR##	footprnt	0	#HEAD#	#TAIL#	250

In module created from: ##FILE## (unrolled) (created)
In namespace "", we have: {
  0 parameter-collections
  0 instantiation-collections
  0 sub-namespaces
  1 definitions
  0 typedefs
  Definitions:
    my_data = user-defined-datatype (defined) my_data
      int<4>
      (
        int<8> x
        bool<> y
      )
      In datatype definition "my_data", we have: {
      Instances:
        fee = bool<> my_data::fee
        fi = bool<> my_data::fi
        fo = bool<> my_data::fo
        fum = bool<> my_data::fum^1
        x = int<8> my_data::x
        y = bool<> my_data::y
      unroll sequence: 
        int<8> x
        bool<> y
        bool<> fee
        bool<> fi
        bool<> fo
        bool<> fum[0..1]
      }

}

footprint: {
}
