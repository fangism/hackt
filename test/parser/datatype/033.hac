// "033.hac"
// copy-modified from "031.hac"
// testing declarations in datatypes
// should reject: no local instances in deftype

deftype my_data <: int<4> (int<8> x; bool y) {
	bool fee, fi, fo, fum[2];
	set { }
	get { }
}

