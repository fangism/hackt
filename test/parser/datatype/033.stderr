At: "033.hac":7:
parse error: syntax error
parser stacks:
state	value
#STATE#	(null) 
#STATE#	(process-prototype) [6:1..44]
#STATE#	{ [6:46]
#STATE#	datatype: bool [7:2..5]
in state #STATE#, possible rules are:
	defdatatype : declare_datatype_proto '{' . optional_datatype_body set_body_optional get_body_optional '}'  (#RULE#)
	optional_datatype_body : .  (#RULE#)
acceptable tokens are: 
	'!' (shift)
	'(' (shift)
	'-' (shift)
	'[' (shift)
	'{' (shift)
	'}' (reduce)
	'~' (shift)
	ID (shift)
	FLOAT (shift)
	INT (shift)
	STRING (shift)
	SCOPE (shift)
	SPEC_LANG (shift)
	SET (reduce)
	GET (reduce)
	BOOL_TRUE (shift)
	BOOL_FALSE (shift)
